### ISSUE TITLE
[XSLT] Built-in function changes to support default values

### ISSUE THREAD

**Issue by rhdunn** [qt4cg/qtspecs#69]
This issue tracks the changes needed to the built-in functions to allow them to combine the declarations into a single definition with default parameter values.

The general approach to this is to make required arguments optional if they are for a function signature that is not the lowest argument count signature, and move any associated logic into the function.

## fn:document
1. Change `fn:document`/`$base-node` from `node()` to `node()?`.

```
- If $base-node is supplied,
+ If $base-node is not empty,
```

## fn:function-available
1. Change `fn:function-available`/`$arity` from `xs:integer` to `xs:integer?`.

If $arity is the empty sequence, the function-available function returns true if and only if there is at least one available function (with some arity) whose name matches the value of the $name argument.

If $arity is not the empty sequence, the function-available function returns true if and only if there is an available function whose name matches the value of the $function-name argument and whose arity matches the value of the $arity argument.

**Comment by ChristianGruen** [comment]
@rhdunn I can’t really judge (as it affects XSLT functions): Is this still valid?

**Comment by michaelhkay** [comment]
Good catch, yes, we've made this change across all the F+O functions but it still needs to be made to XSLT functions.

**Comment by ndw** [comment]
What's above are just examples, the problem is likely a little wider.
